tdh/resource_user.go

```diff
@@ -194,23 +205,57 @@ func (r *userResource) Create(ctx context.Context, req resource.CreateRequest, r
 
 	// Generate API request body from plan
 	userRequest := customer_metadata.CreateUserRequest{
-		Usernames:    []string{plan.Email.ValueString()},
-		ServiceRoles: rolesReq,
+		Usernames: []string{plan.Email.ValueString()},
 	}
+	if r.client.Root.IsSre {
+		plan.Organizations.ElementsAs(ctx, &userRequest.Organizations, true)
+	}
+	if (r.client.Root.IsSre && !plan.Organizations.IsNull()) || !r.client.Root.IsSre {
+		userRequest.ServiceRoles = rolesReq
+		plan.PolicyIds.ElementsAs(ctx, &userRequest.PolicyIds, true)
+	}
+
 	plan.Tags.ElementsAs(ctx, &userRequest.Tags, true)
-	plan.PolicyIds.ElementsAs(ctx, &userRequest.PolicyIds, true)
-	if err := r.client.CustomerMetadata.CreateUser(&userRequest); err != nil {
-		resp.Diagnostics.AddError(
-			"Submitting request to create User",
-			"Could not create User, unexpected error: "+err.Error(),
-		)
-		return
+
+	if r.client.Root.IsSre {
+		if err := r.client.CustomerMetadata.CreateOrgOrSREUser(&userRequest); err != nil {
+			if plan.Organizations.IsNull() {
+				resp.Diagnostics.AddError(
+					"Submitting request to create an Organization User",
+					"Could not create Organization User, unexpected error: "+err.Error(),
+				)
+			} else {
+				resp.Diagnostics.AddError(
+					"Submitting request to create SRE User",
+					"Could not create SRE User, unexpected error: "+err.Error(),
+				)
+			}
+			return
+		}
+	} else {
+		if err := r.client.CustomerMetadata.CreateUser(&userRequest); err != nil {
+			resp.Diagnostics.AddError(
+				"Submitting request to create User",
+				"Could not create User, unexpected error: "+err.Error(),
```

Comment:
keep this common message for error, our resource is only 'user', so no headache for org or normal user... let it fall on provider users 😁